rocks-db/src/columns/offchain_data.rs

```diff
@@ -0,0 +1,138 @@
+use crate::column::TypedColumn;
+use crate::generated::offchain_data_generated::off_chain_data as fb;
+use crate::key_encoders::{decode_string, encode_string};
+use crate::{Result, ToFlatbuffersConverter};
+use entities::models::OffChainDataGrpc;
+use flatbuffers::{FlatBufferBuilder, WIPOffset};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, Clone, Default)]
+pub enum StorageMutability {
+    #[default]
+    Immutable,
+    Mutable,
+}
+
+impl StorageMutability {
+    pub fn is_mutable(&self) -> bool {
+        match self {
+            StorageMutability::Immutable => false,
+            StorageMutability::Mutable => true,
+        }
+    }
+}
+
+impl From<&str> for StorageMutability {
+    fn from(storage_mutability: &str) -> Self {
+        if storage_mutability.is_empty()
+            || storage_mutability.starts_with("ipfs")
+            || storage_mutability.starts_with("arweave")
+        {
+            return StorageMutability::Immutable;
+        } else {
+            return StorageMutability::Mutable;
+        }
+    }
+}
+
+impl From<OffChainData> for OffChainDataGrpc {
+    fn from(off_chain_data: OffChainData) -> Self {
+        OffChainDataGrpc {
+            url: off_chain_data.url.unwrap_or_default(),
+            metadata: off_chain_data.metadata.unwrap_or_default(),
+        }
+    }
+}
+
+impl From<OffChainDataGrpc> for OffChainData {
+    fn from(off_chain_data: OffChainDataGrpc) -> Self {
+        let storage_mutability = StorageMutability::from(off_chain_data.url.as_str());
+        OffChainData {
+            storage_mutability,
+            url: Some(off_chain_data.url),
+            metadata: Some(off_chain_data.metadata),
+            last_read_at: 0,
```

Comment:
lets add a todo to add it to grpc if we get to populating it in p2p

Reply:
Shouldn't each peer keep it's own "last read at" info?

Reply:
It really depends on the level of trust to those peers. Indicating the last checked time over grpc when sharing that data seems a valid thing to do.